src/pybind/mgr/dashboard/frontend/src/app/shared/api/user.service.ts

```diff
@@ -40,6 +58,14 @@ export class UserService {
     });
   }
 
+  validateUserName(user: string): Observable<boolean> {
+    return this.list().pipe(
+      mergeMap((resp: any[]) => {
+        return observableOf(undefined !== resp.find((x) => x.username === user));
+      })
+    );
+  }
```

Comment:
I'm curious... Instead doing a `GET /api/user`, retrieving all users, and then filtering that to find a specific username... Why not using the `GET /api/user/:username` API call directly?

Reply:
Mainly because while calling the `GET /api/user/:username` if the given username doesn't exist a 404 error logs in the console. I tried some ways to handle the error but couldn't. Then I went ahead and used this method. I'll try more and get this done using the `username` API call.

Reply:
Mmmm, interesting. :thinking:  There's a not very commonly used HTTP verb, which is [`HEAD`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD), which is often used to check whether some resource exists without needing the actual payload of that resource (`HEAD` does not return anything in the HTTP body).

However, that will still return a 404, so to avoid having a pop-over, you should modify (I think) api-interceptor.serivce not to generate notifications on HEAD requests.

Reply:
Let me check up on this method, thanks :)

Reply:
@epuertat The main side effect of returning the 404 error is that even if we try to ignore the notification (by specifying it in the api.interceptor.service), the **green tick** in the Username field of the form which indicates the specified username can be used wont be showing. I ran several tests today and every way I tried I'm stumbling into this issue.

Reply:
Mmm, that doesn't make sense... as long as you define the catch error return. Is that the case?

Reply:
The HttpInterceptor will catch the error and returns it. And even then this is the case. Do I need to catch and throw the error in the user-service file? @epuertat

Reply:
Mmm, could you please push your changes? You may check [this suggestion](https://stackoverflow.com/a/49047764/13173750).

Reply:
I pushed in a change which I did for testing purposes. I wrote it in a way to skip the Interceptor but its not detecting the HttpHeader specified in the get() function. Could you check if what I did is correct? @epuertat (Thanks for the suggestion)